src/ui-kit/components/actions/actions-dropdown/actions-dropdown.component.ts

```diff
@@ -75,26 +82,50 @@ export class SamActionsDropdownComponent {
       this.toggleActions();
       event.preventDefault();
       event.stopPropagation();
-    } else if (KeyHelper.is("down",event)){
+    } else {
       this.keyDownHandler(event);
     }
   }
 
+  setFocusOnFirstItem() {
+    this.actionsList.changes.subscribe(t => {
+      this.ngForRendered();
+    })
+  }
+
+  ngForRendered() {
+    if(this.actionsList.length > 0) {
+      this.focusIndex = 0;
+      this.actionsList.toArray()[this.focusIndex].nativeElement.focus();
+    }
+  }
+
   keyDownHandler(event){
     if(KeyHelper.is("down",event)){
       if(this.focusIndex+1<this.actionsList.toArray().length){
         this.focusIndex++;
         this.actionsList.toArray()[this.focusIndex].nativeElement.focus();
+      } else {
+        this.focusIndex = 0;
+        this.actionsList.toArray()[this.focusIndex].nativeElement.focus();
       }
       event.preventDefault();
       event.stopPropagation();
     } else if(KeyHelper.is("up",event)){
       if(this.focusIndex-1>=0){
         this.focusIndex--;
         this.actionsList.toArray()[this.focusIndex].nativeElement.focus();
+      } else {
+        this.focusIndex = this.actionsList.toArray().length - 1;
+        this.actionsList.toArray()[this.focusIndex].nativeElement.focus();
       }
       event.preventDefault();
       event.stopPropagation();
+    } else if (KeyHelper.is("esc", event)){
```

Comment:
Can you add type information for the function parameters and the return type?

Reply:
Which function parameters are you referring to? @yerramshilpa

Reply:
I am refering to if (KeyHelper.is("esc", event)) condition check, can you changes some like below.

import { KeyHelper, KEYS } from '../../utilities/key-helper/key-helper';
if (KeyHelper.is(KEYS.SPACE, evt))

Reply:
Done